### PR TITLE
ReactiveSwift 4.0.

### PR DESCRIPTION
**Pull request by andersio**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 machine:
   xcode:
-    version: "9.0"
+    version: "9.4.1"
```
Comment (renatorodrigues) [inline circle.yml]: Why not Xcode 10 straightaway?

Comment (renatorodrigues) [inline circle.yml]: Why not Xcode 10 straightaway?

**Review by renatorodrigues — APPROVED** [review]

**Review by diegopetrucci — APPROVED** [review]

File: Podfile.lock [hunk 2]
```diff
@@ -1,21 +1,33 @@
...
-  Result: d2d07204ce72856f1fd9130bbe42c35a7b0fea10
+  ReactiveCocoa: 87cb64d78424d749a0bbfd9b07e4825e8d834066
+  ReactiveSwift: a2bb9ace428a109e9c0209615645d9d286c8c433
+  Result: 7645bb3f50c2ce726dd0ff2fa7b6f42bbe6c3713
 
-PODFILE CHECKSUM: dc23c34506ccaf0f3eb382df20e83f60f9a00acc
+PODFILE CHECKSUM: cb3e36c06f0aad276b3bae5804f78b4c467c2030
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.5.3
```
Comment (sergdort) [inline Podfile.lock:33]: Do we have the `gemfile`?

File: ReactiveFeedbackTests/SystemTests.swift [hunk 3]
```diff
@@ -50,10 +50,10 @@ class SystemTests: XCTestCase {
 
     func test_reduce_with_two_immediate_feedback_loops() {
         let feedback1 = Feedback<String, String> { state in
-            return SignalProducer(value: "_a")
+            return !state.hasSuffix("_a") ? SignalProducer(value: "_a") : .empty
         }
         let feedback2 = Feedback<String, String> { state in
-            return SignalProducer(value: "_b")
+            return !state.hasSuffix("_b") ? SignalProducer(value: "_b") : .empty
```
Comment (sergdort) [inline ReactiveFeedbackTests/SystemTests.swift:56]: Why is this changed?

**Review by mluisbrown — APPROVED** [review]

File: ReactiveFeedbackTests/SystemTests.swift (hunk 3 above)
Comment (andersio) [reply ReactiveFeedbackTests/SystemTests.swift:56]: It has been broken since ReactiveSwift 3.1, which fixed a bug regarding constant producer not being interrupted as expected. So with ReactiveSwift 3.1, the events from the second feedback would never reach the reducer, hence the addition of the condition.
